### DISCUSSION TITLE
Backport Offhand from 1.9+

### DISCUSSION THREAD

**Discussion by Irgendwer01** [Roadhog360/Et-Futurum-Requiem#49] [Ideas]
It would be nice if it would be possible that the Offhand get's backported to 1.7.10 like the other Mechanics that got backported and thanks for developing such a nice mod for 1.7.10!

**Comment by Roadhog360** [comment]
> thanks for developing such a nice mod for 1.7.10!

Thank Ganymedes01 for starting it all.

**Comment by Roadhog360** [comment]
Decided to remove "won't add" because I have permission to reference & use Backhand's code.

**Comment by Pilzinsel64** [comment]
Maybe use a Baubles slot for that, if installed. The current version of the Backhand mod just wont show the additional slot. Probably Baubles overwrites it somehow. It's required to use an inventory slot (well, finally haven't installed becauase of that).

**Reply by DarianLStephens** [reply]
The slot was removed because of problems overriding the inventory with other mods.
Perhaps the inventory could be modified directly with mixins, or some 'overlay' of the slot can be applied whenever the player inventory is open, regardless of where the GUI is from?

**Reply by Pilzinsel64** [reply]
Ah, I see. Didn't know that. xD
Well, not sure about how to add an overlay (I am a noob regarding mc gui coding xD) but I can try porting the backhand mod into EFR, if you @Roadhog360 agree. I can also try via a config to use a baubles slot. Not sure about what's possible with the baubles API but I would just check how roadhog did the elytra thing and experiment with the baubles API.

**Reply by Roadhog360** [reply]
Baubles slot should be optional; I don't want it to be a hard dependency. I think @makamys would probably know if an "overlay slot" is feasible. I also didn't add in Backhand because if I recall correctly it still has some outstanding issues that would need to be fixed. If it was ported into EFR it would likely be an experimental feature.

**Reply by Roadhog360** [reply]
Oh and @jss2a98aj would probably be able to help with baubles stuff. I'll ask now if he's willing to give some info; he did actually a lot of the baubles support for me.

**Reply by Pilzinsel64** [reply]
Sounds like I could add that actual Backhand state with optional baubles support. If there are bugs or any improvement, we can always make them at any time.

**Reply by makamys** [reply]
A global overlay slot isn't really feasible, or even desired. Not even vanilla works that way (e.g. you can't see the offhand slot in the anvil GUI in 1.12.2). Theoretically it's possible, as TrashSlot has shown, but it works by having you manually drag the slot around to a place where it looks okay for each GUI.

Doing it for a pre-defined set of GUIs though is definitely doable, which is what Satchels does. Backhand did it in an intrusive way where it replaces the entire player inventory container rather than adding to it via mixins like Satchels, so there's plenty of room for improvement there.

**Reply by DarianLStephens** [reply]
I meant 'global' as in 'applies to all instances of the player inventory', not applying to all GUIs.
Anything that lets you open your inventory, not as part of another GUI. To avoid the need to replace the player inventory with one that has the slot, you just have the overlay instead.

The second part of what you said is closer to what I meant. You could define classes from mods that are replacements to the inventory and have the slot appear there, or at least appear in all instances of the default player inventory, so it can display alongside Satchels as mentioned.

**Reply by Roadhog360** [reply]
This is unfortunately not possible, that is not how GUIs in Minecraft work.

Not sure of desired vanilla solutions at this time but I do think making offhand require baubles to start wouldn't be bad.
